micromanager_gui/main_window.py

```diff
@@ -114,7 +120,10 @@ def __init__(self, viewer: napari.viewer.Viewer, remote=True):
         self.objectives_cfg = None
 
         # create connection to mmcore server or process-local variant
-        self._mmc = RemoteMMCore() if remote else CMMCorePlus()
+        if mmc:
```

Comment:
```suggestion
        if mmc is not None:
```
rarely (though likely not in this case), objects have odd `__bool__` behavior that makes it a bit safer to explicitly check `not None`